near-sdk/src/store/lookup_map/entry.rs

```diff
@@ -0,0 +1,379 @@
+use crate::utils::CacheEntry;
+
+/// A view into a single entry in the map, which can be vacant or occupied.
+pub enum Entry<'a, K: 'a, V: 'a> {
+    Occupied(OccupiedEntry<'a, K, V>),
+    Vacant(VacantEntry<'a, K, V>),
+}
+
+impl<'a, K, V> Entry<'a, K, V>
+where
+    K: Ord,
+{
+    /// Returns a reference to this entry's key.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use near_sdk::store::LookupMap;
+    ///
+    /// let mut map: LookupMap<String, u32> = LookupMap::new(b"m");
+    /// assert_eq!(map.entry("poneyland".to_string()).key(), &"poneyland");
+    /// ```
+    pub fn key(&self) -> &K {
+        match self {
+            Entry::Occupied(entry) => entry.key(),
+            Entry::Vacant(entry) => entry.key(),
+        }
+    }
+
+    /// Ensures a value is in the entry by inserting the default if empty, and returns
+    /// a mutable reference to the value in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use near_sdk::store::LookupMap;
+    ///
+    /// let mut map: LookupMap<String, u32> = LookupMap::new(b"m");
+    ///
+    /// map.entry("poneyland".to_string()).or_insert(3);
+    /// assert_eq!(map["poneyland"], 3);
+    ///
+    /// *map.entry("poneyland".to_string()).or_insert(10) *= 2;
+    /// assert_eq!(map["poneyland"], 6);
+    /// ```
+    pub fn or_insert(self, default: V) -> &'a mut V {
+        match self {
+            Self::Occupied(entry) => entry.into_mut(),
+            Self::Vacant(entry) => entry.insert(default),
+        }
```

Comment:
This can delegate to `or_insert_with`

```suggestion
        self.or_insert_with(|| default)
```

That way, there's a single place which contains the match.

Reply:
same here as above, I'm following std::HashMap pattern. I'll try to dig into comparing what the two compile to exactly to make sure there is no difference